cmd/commands/product-release.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2024 The Codefresh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/codefresh-io/cli-v2/pkg/util"
+
+	"github.com/codefresh-io/go-sdk/pkg/client"
+	platmodel "github.com/codefresh-io/go-sdk/pkg/model/promotion-orchestrator"
+
+	"github.com/spf13/cobra"
+)
+
+type (
+	Slice struct {
+		Edges    []*Edge              `json:"edges"`
+		PageInfo *platmodel.SliceInfo `json:"pageInfo"`
+	}
+
+	Edge struct {
+		Node map[string]any `json:"node"`
+	}
+)
+
+func NewProductReleaseCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "product-release",
+		Short:             "Manage product releases of Codefresh account",
+		PersistentPreRunE: cfConfig.RequireAuthentication,
+		Args:              cobra.NoArgs, // Workaround for subcommand usage errors. See: https://github.com/spf13/cobra/issues/706
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.HelpFunc()(cmd, args)
+			exit(1)
+		},
+	}
+
+	cmd.AddCommand(NewProductReleaseListCommand())
+
+	return cmd
+}
+
+func NewProductReleaseListCommand() *cobra.Command {
+
+	var (
+		productName       string
+		page              int
+		pageLimit         int
+		statusList        string
+		promotionFlowList string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all the pipelines",
+		Args:  cobra.NoArgs,
+		Example: util.Doc(`
+			<BIN> product-release list --product <product>
+		`),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := cmd.Context()
+
+			releaseStatus, err := ToProductReleaseStatus(statusList)
+			promotionFlows, err := ToPromotionFlows(promotionFlowList)
+			if err != nil {
+				return err
+			}
+			filterArgs := platmodel.ProductReleaseFiltersArgs{
+				Statuses:       releaseStatus,
+				PromotionFlows: promotionFlows,
+			}
+			return RunProductReleaseList(ctx, filterArgs, productName, page, pageLimit)
+		},
+	}
+
+	cmd.Flags().StringVarP(&statusList, "status", "s", "", "Filter by statuses")
+	cmd.Flags().StringVar(&promotionFlowList, "promotion-flows", "", "Filter by promotion flows")
+	cmd.Flags().IntVar(&page, "page", 1, "page number")
+	cmd.Flags().IntVar(&pageLimit, "page-limit", 20, "page limit number")
+	cmd.Flags().StringVarP(&productName, "product", "p", "", "product")
```

Comment:
where are the start and end date filters?

Reply:
yeah we dont use them in the end I just forgot to generate the docs, sorry for confusion

Reply:
and why don't i see pageLimit in the md file?